Fix dev selector typo

teresaromero opened the pull request:
Fixes #2778 

## Proposed changes

- Separate selection logic from GetDevFromManifest. This function will though an error (ErrNoDevSelected) if devName is empty and manifest has multiple devs defined.
- new SelectDevFromManifest fallback function when err is ErrNoDevSelected, where label is a param that represents the question of the selector

jLopezbarb commented on cmd/utils/dev_test.go:
```diff
@@ -310,6 +310,29 @@ func Test_GetDevFromManifest(t *testing.T) {
 			err:     nil,
 		},
+		{
+			name: "manifest has one dev section and user intrudices empty devName",
```
```suggestion
			name: "manifest has one dev section and user introduces empty devName",
```

jLopezbarb commented on cmd/utils/dev_test.go:
```diff
@@ -310,6 +310,29 @@ func Test_GetDevFromManifest(t *testing.T) {
+			err:     nil,
+		},
+		{
+			name: "manifest has several dev section user introduces emtpy devName",
```
```suggestion
			name: "manifest has several dev section user introduces empty devName",
```

ifbyol commented on cmd/utils/dev.go:
```diff
@@ -162,19 +168,24 @@ func GetDevFromManifest(manifest *model.Manifest, devName string) (*model.Dev, e
 	}
+}
+
+func SelectDevFromManifest(manifest *model.Manifest, label string) (*model.Dev, error) {
```
Can you include a comment in this exported function, please?

ifbyol reviewed (commented):
@rlamana @RinkiyaKeDad Can you review the text for the elector on each situation, please?

RinkiyaKeDad commented on cmd/doctor.go:
```diff
@@ -65,7 +66,13 @@ func Doctor() *cobra.Command {
+				if !errors.Is(err, utils.ErrNoDevSelected) {
+					return err
+				}
+				dev, err = utils.SelectDevFromManifest(manifest, "Select the development container you want download logs:")
```
```suggestion
				dev, err = utils.SelectDevFromManifest(manifest, "Select which development container's logs to download:")
```

RinkiyaKeDad commented on cmd/down.go:
```diff
@@ -78,7 +79,13 @@ func Down() *cobra.Command {
+					if !errors.Is(err, utils.ErrNoDevSelected) {
+						return err
+					}
+					dev, err = utils.SelectDevFromManifest(manifest, "Select the development container you want to deactivate:")
```
```suggestion
					dev, err = utils.SelectDevFromManifest(manifest, "Select which development container to deactivate:")
```

RinkiyaKeDad commented on cmd/exec.go:
```diff
@@ -191,15 +192,17 @@ func getDevFromArgs(manifest *model.Manifest, args []string) (*model.Dev, error)
 		}
-	} else {
-		dev, err = utils.GetDevFromManifest(manifest, args[0])
+		dev, err = utils.SelectDevFromManifest(manifest, "Select the development container where you want to exec:")
```
```suggestion
		dev, err = utils.SelectDevFromManifest(manifest, "Select which development container to exec:")
```

RinkiyaKeDad commented on cmd/push.go:
```diff
@@ -98,7 +99,13 @@ func Push(ctx context.Context) *cobra.Command {
+				if !errors.Is(err, utils.ErrNoDevSelected) {
+					return err
+				}
+				dev, err = utils.SelectDevFromManifest(manifest, "Select the development container where you want to push:")
```
```suggestion
				dev, err = utils.SelectDevFromManifest(manifest, "Select which development container to push to:")
```

RinkiyaKeDad commented on cmd/restart.go:
```diff
@@ -58,7 +59,13 @@ func Restart() *cobra.Command {
+				if !errors.Is(err, utils.ErrNoDevSelected) {
+					return err
+				}
+				dev, err = utils.SelectDevFromManifest(manifest, "Select the development container you want to restart:")
```
```suggestion
				dev, err = utils.SelectDevFromManifest(manifest, "Select which development container to restart:")
```

RinkiyaKeDad commented on cmd/status.go:
```diff
@@ -62,7 +63,13 @@ func Status() *cobra.Command {
+				if !errors.Is(err, utils.ErrNoDevSelected) {
+					return err
+				}
+				dev, err = utils.SelectDevFromManifest(manifest, "Select the development container you want to get sync status:")
```
```suggestion
				dev, err = utils.SelectDevFromManifest(manifest, "Select which development container's sync status is needed:")
```

RinkiyaKeDad commented on cmd/up/up.go:
```diff
@@ -262,7 +262,13 @@ func Up() *cobra.Command {
+				if !errors.Is(err, utils.ErrNoDevSelected) {
+					return err
+				}
+				dev, err = utils.SelectDevFromManifest(oktetoManifest, "Select the development container you want to activate:")
```
```suggestion
				dev, err = utils.SelectDevFromManifest(oktetoManifest, "Select which development container to activate:")
```

RinkiyaKeDad reviewed (commented):
I've added suggestions for the messages. 

Another thing I was thinking was that we could just change all the messages to "Select a development container:" - since the user running the command would know from the command itself what exactly they are selecting the container for. What do you all think?

teresaromero commented:
> I've added suggestions for the messages.
> 
> Another thing I was thinking was that we could just change all the messages to "Select a development container:" - since the user running the command would know from the command itself what exactly they are selecting the container for. What do you all think?

Thnks for the suggestions! I've fixed the strings accordingly. 

Regarding the message, i think adding the action that is being performed, although the user knows from the command, is more descriptive.
